src/devices/bus/rs232/qvt102.cpp

```diff
@@ -0,0 +1,87 @@
+// license:BSD-3-Clause
+// copyright-holders:68bit
+
+#include "emu.h"
+#include "qvt102.h"
+
+qvt102_terminal_device::qvt102_terminal_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, SERIAL_TERMINAL_QVT102, tag, owner, clock)
+	, device_rs232_port_interface(mconfig, *this)
+	, m_qvt102(*this, "qvt102")
+	, m_flow_control(*this, "FLOW_CONTROL")
+{
+}
+
+void qvt102_terminal_device::device_add_mconfig(machine_config &config)
+{
+	QVT102(config, m_qvt102, 0);
+	m_qvt102->rs232_conn_txd_handler().set(FUNC(qvt102_terminal_device::output_rxd));
+	m_qvt102->rs232_conn_rts_handler().set(FUNC(qvt102_terminal_device::route_term_rts));
+	m_qvt102->rs232_conn_dtr_handler().set(FUNC(qvt102_terminal_device::route_term_dtr));
+}
+
+INPUT_PORTS_START(qvt102_terminal)
+
+	PORT_START("FLOW_CONTROL")
+	PORT_CONFNAME(0x1, 1, "Flow Control") PORT_CHANGED_MEMBER(DEVICE_SELF, qvt102_terminal_device, flow_control, 0)
+	PORT_CONFSETTING(0x00, "None")
+	PORT_CONFSETTING(0x01, "Terminal DTR to remote CTS")
+
+INPUT_PORTS_END
+
+ioport_constructor qvt102_terminal_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME(qvt102_terminal);
+}
+
+WRITE_LINE_MEMBER(qvt102_terminal_device::input_txd)
+{
+	m_qvt102->rs232_conn_rxd_w(state);
+}
+
+WRITE_LINE_MEMBER(qvt102_terminal_device::route_term_rts)
+{
+	// Loop the terminal RTS output to the terminal CTS input.
+	m_qvt102->rs232_conn_cts_w(state);
+}
+
+// This terminal uses DTR for hardware flow control.
+WRITE_LINE_MEMBER(qvt102_terminal_device::route_term_dtr)
+{
+	if (m_flow_control->read())
+	{
+		// Connect the terminal DTR output to CTS at the other end.
+		output_cts(state);
+	}
+
+	// Cache the state, in case the ioport setting changes.
+	m_dtr = state;
+}
+
+INPUT_CHANGED_MEMBER(qvt102_terminal_device::flow_control)
+{
+	if (newval)
+		output_cts(m_dtr);
+	else
+		output_cts(0);
+}
+
+void qvt102_terminal_device::device_start()
+{
+	save_item(NAME(m_dtr));
+}
+
+void qvt102_terminal_device::device_reset()
+{
+	// To the terminal
+	m_qvt102->rs232_conn_cts_w(0);
+
+	// To the computer
+	output_rxd(1);
+	output_dcd(0);
+	output_dsr(0);
+	if (!m_flow_control->read())
+		output_cts(0);
```

Comment:
This thing needs to be completely dumb - this can get out of sync with device state.  Don't push constant values here, let the devices on the other sides of the proxy push through their correct reset values.